compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -682,7 +682,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                         bx.cx().const_usize(val)
                     }
                     mir::NullOp::UbChecks => {
-                        let val = bx.tcx().sess.opts.debug_assertions;
+                        let val = bx.tcx().sess.ub_checks();
```

Comment:
Isn't there also an implementation in rustc_const_eval that needs updating?

Also, the docs of the intrinsic in `library/core` likely need updating.